include/psa/crypto_se_driver.h

```diff
@@ -961,6 +961,83 @@ typedef struct {
 
 /**@}*/
 
+/** \defgroup se_registration Secure element driver registration
+ */
+/**@{*/
+
+/** A structure containing pointers to all the entry points of a
+ * secure element driver.
+ *
+ * Future versions of this specification may add extra substructures at
+ * the end of this structure.
+ */
+typedef struct {
+    /** The version of the driver model that this driver implements.
```

Comment:
How does it protect against linking against a different version? Isn't it a runtime mechanism?

Reply:
It's a runtime mechanism that protects against linking errors. Drivers are likely to be built independently from the core. If you link a driver that's designed for a certain HAL version with a core that doesn't support this version, you'll get an error during initialization.

There's probably a way to make this a link-time mechanism, by defining a symbol whose name changes at each version. Can this be done portably, and in a way that's basically transparent when you're writing a driver?